ldap/servers/slapd/pagedresults.c

```diff
@@ -361,6 +365,7 @@ pagedresults_free_one_msgid_nolock(Connection *conn, ber_int_t msgid)
         } else {
             slapi_log_err(SLAPI_LOG_TRACE,
                           "pagedresults_free_one_msgid_nolock", "=> msgid=%d\n", msgid);
+            pthread_mutex_lock(mutex);
```

Comment:
A dummy question not related to that specific line.
The "random" mutex is taken from a hash table based on the 'conn' value (pageresult_lock_get_addr). How can we be sure there is no collision ?
For example, two different connections using the same mutex.

Another question, I believe it is possible to have several paged result searches on the same connection. Any risk they are stepping on each over toes ? I mean lock/unlock the same mutex

Reply:
Indeed in both these scenarios different threads will use the same mutex and there will be some lock contention.
But that is the finality of a mutex and the contention is smaller with an array of mutexes than with a single one !

Reply:
Yes that make sense. The target was to reduce the contention. thanks